packages/validator/src/convertor.ts

```diff
@@ -0,0 +1,98 @@
+ */
+export function convert(specsPath: string | undefined, templatePath: string | undefined): Result {
+  if (!specsPath || !templatePath) {
+    return { valid: false, messages: [logger.error('Specification and template file must be provided')], output: {} };
+  }
+
+  let template, specs;
+
+  try {
+    template = fs.readFileSync(templatePath);
+  } catch (e) {
+    return { valid: false, messages: [logger.error(`Unable to read file ${templatePath}`)], output: {} };
+  }
+
+  try {
+    specs = fs.readFileSync(specsPath);
+  } catch (e) {
+    return { valid: false, messages: [logger.error(`Unable to read file ${specsPath}`)], output: {} };
+  }
+
+  return convertJson(specs, template);
+}
+
+/**
+ * Converts specification from provided string and converts it into format the template specifies
+ * @param specs - specification to convert, root must be an object (not an array)
+ * @param template - template json
+ * @returns array of messages and converted specification
+ */
+export function convertJson(specs: string, template: string): Result {
+  try {
+    const nonRedundant = {};
+    const output = {};
+    const parsedTemplate = JSON.parse(template);
+    const parsedSpecs = JSON.parse(specs);
+
+    const result = processSpecs(parsedSpecs, parsedTemplate, '', nonRedundant, {
+      specs: parsedSpecs,
+      nonRedundantParams: nonRedundant,
+      output,
+    });
+
+    return { valid: result.valid, messages: result.messages, output: JSON.parse(JSON.stringify(output)) };
+  } catch (e) {
+    return { valid: false, messages: [{ level: 'error', message: `${e.name}: ${e.message}` }], output: {} };
+  }
+}
+
+/**
+ * Converts specification from one format into another
+ * @param from - format of the provided specification, can be "oas" or "ois"
+ * @param to - format specification will be converted into, can be "ois" or "cs"
+ * @param specs - specification that will be converted
+ * @returns - converted specification as a Result object
+ */
+export function convertFromTo(from, to, specs): Result {
+  from = from.toLowerCase();
+  to = to.toLowerCase();
+
+  let ois: Result = { valid: true, messages: [], output: {} };
+  let cs: Result = { valid: true, messages: [], output: {} };
+
+  if (from === 'oas') {
+    switch (to) {
+      case 'ois':
+        return convert(specs, oas2ois);
+      case 'cs':
+      case 'configsecurity':
+      case 'configandsecurity':
+        ois = convert(specs, oas2ois);
+        cs = convertJson(ois.output ? JSON.stringify(ois.output) : '{}', fs.readFileSync(ois2cs).toString());
+        return { valid: ois.valid, messages: ois.messages, output: cs.output };
+      default:
+        return { valid: false, messages: [invalidConversionMessage(from, to)] };
+    }
+  } else if (from === 'ois' && (to === 'cs' || to === 'configsecurity' || to === 'configandsecurity')) {
+    return convert(specs, ois2cs);
+  }
+
+  return { valid: false, messages: [invalidConversionMessage(from, to)] };
+}
```

Comment:
I find the structure of this kind of awkward and hard to follow. 

Correct me if I'm wrong, but these are all possible cases (using my own informal notation):

```
oas -> ois
oas -> cs|configsecurity|configandsecurity
ois -> cs|configsecurity|configandsecurity
```
otherwise, return `invalidConversionMessage`.

If so, I would make these possible options more explicit in the structure of this code, i.e.:

```
if (from === 'oas' && to === 'ois') {
    ...
} else if (from === 'oas' && (to === 'cs' || to === 'configsecurity' || to === 'configandsecurity')) {
    ...
} else if (from === 'ois' && (to === 'cs' || to === 'configsecurity' || to === 'configandsecurity')) {
} else {
    return { valid: false, messages: [invalidConversionMessage(from, to)] 
}
```

Reply:
Your possible cases seem correct, there is a table in the readme describing this:
| From | To |
| ----- | -----|
| oas | ois |
| oas | cs / ConfigSecurity / ConfigAndSecurity |
| ois | cs / ConfigSecurity / ConfigAndSecurity |
Simplified in 7e8d08e